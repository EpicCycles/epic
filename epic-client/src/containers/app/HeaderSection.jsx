import React, {Fragment} from "react";
import {Link} from "react-router-dom";

const HeaderSection = props => (
    <li className="dropdown">
        {props.sectionContents.map(section => {
            return (
                <Fragment key={'navSection' + section.sectionPos}>
                    <p className="dropbtn">{section.groupHeader}</p>
                    <div className="dropdown-content">
                        <div className="dropdown-section">{section.groupHeader}</div>
                        {section.groupLinks.map(linkData => {
                            return <Link
                                key={`link${linkData.linkRoute.substring(1)}`}
                                className="dropbtn"
                                to={linkData.linkRoute ? linkData.linkRoute : linkData.linkURL}
                            >
                                {linkData.displayText}
                            </Link>
                        })}
                    </div>
                </Fragment>
            )
        })}
    </li>
);
export default HeaderSection;

