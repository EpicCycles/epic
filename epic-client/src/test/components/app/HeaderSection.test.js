import React from 'react';
import HeaderSection from "../../../components/menus/HeaderSection";

// props are user, application, removeMessage
describe("HeaderSection.index tests", () => {
    it('renders the HeaderSection correctly', () => {
        const sectionContents = [
            {
                groupHeader: "Customer",
                groupLinks: [
                    { displayText: "Find Customer", linkRoute: "/customer-search" },
                    { displayText: "New Customer", linkRoute: "/customer" },
                ]
            },
            {
                groupHeader: "Core Data",
                groupLinks: [
                    { displayText: "Quote Sections", linkRoute: "/framework" },
                    { displayText: "Brands", linkRoute: "/brands" },
                ]
            },
        ];
        const headerSection = shallow(<HeaderSection sectionContents={sectionContents} />);
        expect(headerSection).toMatchSnapshot();
    });

});
