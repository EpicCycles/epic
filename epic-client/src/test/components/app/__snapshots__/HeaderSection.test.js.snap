// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSection.index tests renders the HeaderSection correctly 1`] = `
Array [
  <li
    className="dropdown"
    key="navSectionundefined"
  />,
  <li
    className="dropdown"
    key="navSectionundefined"
  />,
]
`;
